napalm_logs/listener/zmq.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+'''
+ZeroMQ listener for napalm-logs.
+'''
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+# Import stdlib
+import time
+import logging
+
+# Import third party libs
+try:
+    import zmq
+    HAS_ZMQ = True
+except ImportError as err:
+    HAS_ZMQ = False
+
+# Import napalm-logs pkgs
+from napalm_logs.listener.base import ListenerBase
+from napalm_logs.exceptions import ListenerException
+
+log = logging.getLogger(__name__)
+
+
+class ZMQListener(ListenerBase):
+    '''
+    ZMQ listener class.
+    '''
+    def __init__(self, address, port, **kwargs):
+        if kwargs.get('address'):
+            address = kwargs['address']
+        if kwargs.get('port'):
+            port = kwargs['port']
+        self.address = address
+        self.port = port
+        self.hwm = kwargs.get('hwm')
+        self.keepalive = kwargs.get('keepalive', 1)
+        self.keepalive_idle = kwargs.get('keepalive_idle', 300)
+        self.keepalive_interval = kwargs.get('keepalive_interval', -1)
+        self.recvtimeout = kwargs.get('timeout')
+        self.protocol = kwargs.get('protocol', 'tcp')
+        self.type = kwargs.get('socket_type', 'PULL')
+
+    def start(self):
+        '''
+        Startup the kafka consumer.
+        '''
+        zmq_uri = '{protocol}://{address}:{port}'.format(
+                       protocol=self.protocol,
+                       address=self.address,
+                       port=self.port
+                   ) if self.port else\
+                   '{protocol}://{address}'.format(  # noqa
+                       protocol=self.protocol,
+                       address=self.address
+                   )
+        log.debug('ZMQ URI: %s', zmq_uri)
+        self.ctx = zmq.Context()
+        if hasattr(zmq, self.type):
+            skt_type = getattr(zmq, self.type)
+        else:
+            skt_type = zmq.PULL
+        self.sub = self.ctx.socket(skt_type)
+        self.sub.connect(zmq_uri)
+        if self.hwm is not None:
+            try:
+                self.sub.setsockopt(zmq.SNDHWM, self.hwm)
```

Comment:
As this is a sub socket it will not have `SNDHWN`, if it is zmq < 3 it wil lhave `HWM`, so you should check for that instead.

Reply:
Good catch